aptos-move/framework/aptos-framework/sources/voting.move

```diff
@@ -326,6 +339,55 @@ module aptos_framework::voting {
         option::extract(&mut proposal.execution_content)
     }
 
+    /// Resolve the multi-step proposal with given id. Can only be done if there are at least as many votes as min required and
+    /// there are more yes votes than no. If either of these conditions is not met, this will revert.
+    ///
+    /// @param voting_forum_address The address of the forum where the proposals are stored.
+    /// @param proposal_id The proposal id.
+    /// @param next_execution_hash The next execution hash of this multi-step proposal
+    public fun resolve_multi_step_proposal<ProposalType: store>(
+        voting_forum_address: address,
+        proposal_id: u64,
+        curr_execution_hash: vector<u8>,
```

Comment:
Why is this passed as an argument? Shouldn't this be stored in the proposal? If you can pass any curr_execution_hash, wouldn't you be able to execute anything?

Reply:
good point, updating now : )